packages/checkbox/src/Check.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { Transition, TransitionStatus } from 'react-transition-group';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { palette } from '@leafygreen-ui/palette';
+import {
+  checkAnimationDuration,
+  checkBoxSize,
+  rippleScale,
+  rippleTransitionDelay,
+  rippleTransitionScale,
+  insetPct,
+} from './constants';
+import SvgCheck from './SvgCheck';
+import SvgIndeterminate from './SvgIndeterminate';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { CheckProps } from './types';
+
+const checkWrapperBaseStyle = css`
+  --lg-checkbox-border-color: ${palette.gray.dark2};
```

Comment:
This CSS variable is only used in this file. Might be a good time to start discussing the merits of CSS vs JS variables:
```
border: 2px solid var(--lg-checkbox-border-color);
```
vs
```
border: 2px solid ${borderColor};
```

I'm inclined to think we should be using the latter pattern most of the time for a number of reasons:
- Value interpolation over CSS vars keeps the CSS declarations to only property declarations where possible.
- We have to interpolate the value anyways in the variable declaration.
- We're interpolating JS variables elsewhere in this and other blocks.
- Where relevant, we can use polished color transformation functions with JS variables, but not with CSS variables.
- We don't need to worry about the variable being named to avoid collisions (`--lg-checkbox-border-color` vs `borderColor`).
- Finding where a CSS variable is declared can be much harder than finding a JS variable declaration, making it harder to understand what code is doing. JS variables are either declared in the file, or imported from another, making it clear to see what the value is. CSS variables can be effectively declared anywhere, and don't leave a similar trail.
- "Side effects" are built into how CSS variables work. You can't reliably expect the value of a CSS variable to not change, and tracking bugs related to them can be more difficult as a result.

If you're not sold, we could spend some time chatting about this today at the group code review 😄

Reply:
I think it's very much a case-by-case consideration. I think the rule-of-thumb you presented is a fair one, (i.e.  don't use CSS vars that are only used in one file)

In this particular case, I think I would agree that JS interpolation is probably the clearer choice. In particular because of how it's used in `checkWrapperCheckedDisabledStyle` on Line 78. It's not immediately clear what the value for the var will be in this since it depends on the order in which `checkWrapperCheckedStyle` and `checkWrapperDisabledStyle` are passed into `cx` on Line 152 (affecting the cascade). Since we'll always want "checked + disabled" to be gray, it's probably best to hard code it in this circumstance.

Reply:
I think that inheritance is both CSS variables' largest strength, and weakness. If we're utilizing that inheritance heavily, we're simultaneously making it more difficult to understand what's happening – especially if we change the value at any point.

The third alternative, while boring, is to simply share a const between files. A prop or context might just be overcomplicating what is, in this case (I believe), a static variable.

If we wanted to turn this into a case where CSS variables uniquely simplifies a problem, if we were to implement `prefers-reduced-motion`, I believe we could do it universally much more simply here using CSS variables. Which reminds me, we should probably do this! 😄

Reply:
There's no universal check that we do for `prefers-reduced-motion`, but there should be!

Reduced motion is the main reason I'm using CSS vars here in [Checkbox.tsx](https://github.com/mongodb/leafygreen-ui/blob/8f64a85536b4201d113b6d1908ee344a210c9783/packages/checkbox/src/Checkbox.tsx#L39)

Reply:
It might actually be useful to create a hook than a global var:
```
const prefersReducedMotion = useReducedMotion()
...
cx({
  [animationStyles] : !prefersReducedMotion
})
```
where
```
function useReducedMotion() {
  return window.matchMedia('prefers-reduced-motion: reduce').matches
}

Reply:
We already have a function that generates styles that are applied when the setting is:

```js
<div className={prefersReducedMotion('display: none')} />
```

See `a11y/src/prefersReducedMotion`

Reply:
Looking into it more, a context seems like overkill for this since the value will hardly ever change.

I think an imported query selector is a little more usable than the function we have. It follows the Emotion guidance on media queries, and will support syntax highlighting.

e.g.
```js
const reducedMotionQuery = '@media (prefers-reduced-motion: reduce)`
...

const styles = css`
  transition-duration: 100ms;

  ${reducedMotionQuery} {
    transition-duration: 0ms;
  }

`

```